engine/src/main/java/org/terasology/engine/telemetry/Metrics.java

```diff
@@ -50,28 +51,19 @@
 
     public void initialise(Context context) {
 
-        Set<Class> metricsClassSet = fetchMetricsClassFromEnvironemnt(context);
+        Set<Class> metricsClassSet = fetchMetricsClassFromEngineOnlyEnvironment(context);
         initializeMetrics(metricsClassSet, context);
     }
 
-    private Set<Class> fetchMetricsClassFromEnvironemnt(Context context) {
-
+    private Set<Class> fetchMetricsClassFromEngineOnlyEnvironment(Context context) {
         ModuleManager moduleManager = context.get(ModuleManager.class);
         Set<Class> metricsClassSet = new HashSet<>();
-        DependencyResolver resolver = new DependencyResolver(moduleManager.getRegistry());
-        for (Name moduleId : moduleManager.getRegistry().getModuleIds()) {
-            Module module = moduleManager.getRegistry().getLatestModuleVersion(moduleId);
-            if (module.getResources() instanceof DirectoryFileSource) {
-                ResolutionResult result = resolver.resolve(moduleId);
-                if (result.isSuccess()) {
-                    try (ModuleEnvironment environment = moduleManager.loadEnvironment(result.getModules(), false)) {
-                        for (Class<?> holdingType : environment.getTypesAnnotatedWith(TelemetryCategory.class, new FromModule(environment, moduleId))) {
-                            metricsClassSet.add(holdingType);
-                        }
-                    }
-                }
+        try (ModuleEnvironment environment = moduleManager.loadEnvironment(Collections.emptySet(), false)) {
```

Comment:
`loadEnvironment(Collections.emptySet())` is not something I expected to see, but hey, in this current context of "fix the regressions," I'm on board. It gets rid of the "scan the whole registry" thing we wanted to get rid of, but other than that it doesn't change from the previous approach more than necessary.

We already know that this part of the ModuleManager interface is something we want to revisit.